pydicom/filebase.py

```diff
@@ -30,19 +30,13 @@ def __init__(self, *args, **kwargs):
     def read_le_tag(self):
         """Read and return two unsigned shorts (little endian)
            from the file."""
-        bytes_read = self.read(4)
-        if len(bytes_read) < 4:
-            # needed for reading "next" tag when at end of file
-            raise EOFError
+        bytes_read = self.read(4, need_exact_length=True)
         return unpack(b"<HH", bytes_read)
 
     def read_be_tag(self):
```

Comment:
I would do a function decorator. Announcing its deprecation in v1.0. 
If we don't have a deprecation policy, [here's scikit-learn](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/deprecation.py). @glemaitre is familiar with it. 
I would open a separated PR to add the deprecation decorator, and then use it here.